### PR TITLE
[TASK] Retain psalm issues for future review

### PR DESCRIPTION
**Pull request by SignpostMarv**
cherry-picking the commits from #778 that pertain to issues that either need to be retained until resolved by discussion on other issues, or are being retained as a result of said discussion.

### CODE REVIEW

**Comment by JakeQZ** [comment]
I've added labels and edited the title to indicate this contains commits for reference, and is not something to be reviewed and merged right now - as I understand that to be the case.

**Comment by SignpostMarv** [comment]
this particular PR isn't a WIP, it's to flag issues that shouldn't be removed in PRs without first checking other issues
* `NonStaticSelfCall` - existing issues will likely be permanently retained, pending outcome of #792 
* `RedundantConditionGivenDocblockType` - some issues are being permanently retained, as a result of discussion in #537
* flagged issues for `src/Utilities/CssConcatenator.php` are probably under the purview of #742, so shouldn't be resolved unless specifically as part of that issue.

**Comment by JakeQZ** [comment]
> this particular PR isn't a WIP, it's to flag issues that shouldn't be removed in PRs without first checking other issues

OK.  Maybe I could have chosen a better title prefix.  I just wanted to make it clear that

> SignpostMarv wants to merge 4 commits into MyIntervals:master from SignpostMarv:retain-psalm-issues-for-future-review

is misleading.

**Comment by SignpostMarv** [comment]
> > SignpostMarv wants to merge 4 commits into MyIntervals:master from SignpostMarv:retain-psalm-issues-for-future-review
> 
> is misleading.

ah, clarification; this is in relation to #793, as running `psalm --set-baseline=./psalm.baseline.xml` after upgrading psalm may remove baseline entries due to bugfixes in later releases of psalm (i.e. you'd be going from 3.2.12 to ^3.5), or because issues were reclassified by psalm so the comments would need to be added elsewhere, as opposed to tapping `[s]` when `git commit -p`

**Comment by SignpostMarv** [comment]
"future review" in this instance meaning "when upgrading psalm / project to php 7.1"

**Comment by JakeQZ** [comment]
OK.  I'm still confused as to why `psalm.baseline.xml` needs to be in the repo at all - it looks more like a logfile to me?  Or does it also act as a cache to speed up future runs?  (I note I also have a file called `.php_cs.cache` which is not in the repo.)  But like I said, I'm new to Psalm 😕

**Comment by SignpostMarv** [comment]
`psalm.xml` is the config, where you can set types of issues to be set to globally `info` or `suppress`. one can also set individual files or directories to be ignored.

`psalm.baseline.xml` keeps a record of *specific instances* of issues, effectively setting those instances to `info`.

travis runs `psalm --show-info=false`, passing the psalm check if no new issues are introduced.

`psalm --update-baseline` effectively removes issues that're resolved **without recording new issues introduced**

`psalm --set-baseline=[path-to-file]` records all current issues, removing issues that're resolved by virtue of them not being included in the output.

any issue not currently referenced in the baseline will necessarily be "new" and will cause travis to fail, issues which might not be caught if the baseline were not employed and `<issueHandlers>` or `<ignoreFiles`> were used instead.

tl:dr; baseline is more of a record than a log and helps identify newly introduced issues.

**Comment by SignpostMarv** [comment]
I'll not be able to work on this while I'm sat at the day job, but could this be merged for this evening ?

**Comment by SignpostMarv** [comment]
thanks :D

**Comment by oliverklee** [comment]
You're welcome. Thanks for working on the psalm issues! :coffee:

**Comment by JakeQZ** [comment]
> 
> 
> I've added labels and edited the title to indicate this contains commits for reference, and is not something to be reviewed and merged right now - as I understand that to be the case.

Apologies.  It seems I completely misunderstood this PR.

I ***will*** get up to speed with Psalm, I promise...

🥧(humble)
